authority/provisioner/acme_test.go

```diff
@@ -88,92 +90,285 @@ func TestACME_Init(t *testing.T) {
+		p     *ACME
+		token string
+		code  int
+		err   error
+	}
+	tests := map[string]func(*testing.T) test{
+		"not-implemented": func(t *testing.T) test {
+			p, err := generateACME()
+			assert.FatalError(t, err)
+			return test{
+				p:     p,
+				token: "foo",
+				code:  http.StatusUnauthorized,
+				err:   errors.New("provisioner.AuthorizeSSHSign not implemented"),
+			}
+		},
+	}
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			tc := tt(t)
+			opts, err := tc.p.AuthorizeSSHSign(context.Background(), tc.token)
+			if assert.NotNil(t, err) {
+				if assert.NotNil(t, tc.err) {
+					assert.Nil(t, opts)
+					sc, ok := err.(errs.StatusCoder)
+					assert.Fatal(t, ok, "error does not implement StatusCoder interface")
+					assert.Equals(t, sc.StatusCode(), tc.code)
+					assert.HasPrefix(t, err.Error(), tc.err.Error())
+				}
+			}
+		})
+	}
+}
+
+func TestACME_AuthorizeSSHRevoke(t *testing.T) {
+	type test struct {
+		p     *ACME
+		token string
+		code  int
+		err   error
+	}
+	tests := map[string]func(*testing.T) test{
+		"not-implemented": func(t *testing.T) test {
+			p, err := generateACME()
+			assert.FatalError(t, err)
+			return test{
+				p:     p,
+				token: "foo",
+				code:  http.StatusUnauthorized,
+				err:   errors.New("provisioner.AuthorizeSSHRevoke not implemented"),
+			}
+		},
+	}
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			tc := tt(t)
+			err := tc.p.AuthorizeSSHRevoke(context.Background(), tc.token)
+			if assert.NotNil(t, err) {
+				if assert.NotNil(t, tc.err) {
+					sc, ok := err.(errs.StatusCoder)
+					assert.Fatal(t, ok, "error does not implement StatusCoder interface")
+					assert.Equals(t, sc.StatusCode(), tc.code)
+					assert.HasPrefix(t, err.Error(), tc.err.Error())
+				}
+			}
+		})
+	}
+}
+
+func TestACME_AuthorizeSSHRenew(t *testing.T) {
+	type test struct {
+		p     *ACME
+		token string
+		code  int
+		err   error
+	}
+	tests := map[string]func(*testing.T) test{
+		"not-implemented": func(t *testing.T) test {
+			p, err := generateACME()
+			assert.FatalError(t, err)
+			return test{
+				p:     p,
+				token: "foo",
+				code:  http.StatusUnauthorized,
+				err:   errors.New("provisioner.AuthorizeSSHRenew not implemented"),
+			}
+		},
+	}
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			tc := tt(t)
+			cert, err := tc.p.AuthorizeSSHRenew(context.Background(), tc.token)
+			if assert.NotNil(t, err) {
+				if assert.NotNil(t, tc.err) {
+					assert.Nil(t, cert)
+					sc, ok := err.(errs.StatusCoder)
+					assert.Fatal(t, ok, "error does not implement StatusCoder interface")
+					assert.Equals(t, sc.StatusCode(), tc.code)
+					assert.HasPrefix(t, err.Error(), tc.err.Error())
+				}
+			}
+		})
+	}
+}
+
+func TestACME_AuthorizeSSHRekey(t *testing.T) {
+	type test struct {
+		p     *ACME
+		token string
+		code  int
+		err   error
+	}
+	tests := map[string]func(*testing.T) test{
+		"not-implemented": func(t *testing.T) test {
```

Comment:
@dcow: fyi I usually write tests using a [gotests](https://github.com/cweill/gotests), given a method it automatically generates a [table driven test](https://github.com/golang/go/wiki/TableDrivenTests) looking like:
```go
tests := []struct {
        name string
        foo string
        bar int
        want int
        wantErr bool
}{
    {"test name", "foo", 1, "200", false},
    {"test name 2", "bar", 2, "500", true},
    ...
}
for _, tt := range tests {
    ...
}
```

Reply:
You can integrate it in your editor and it will help you a lot, sometimes you will need to change what it gives you by default, but you can use exactly that for most of the tests.

Reply:
I think mariano and I have had this discussion before, but there are a few reasons I don't love this specific style: 

1) You're not testing for a specific error, which means something in the code might change and an error might be thrown earlier in the execution, but your test wouldn't find that.
2) A lot of our tests require some prefabricating of objects, tokens, authorities, etc. that are then used in the test. Using this style you can end up with a large piece of the test that defines all the objects used later on in the actual tests struct. It makes it difficult to backtrack and change things.

Just my opinions, which is why my tests tend to look different than marianos 🤷‍♂

Reply:
you can always use `wantErr error`

Reply:
I think my comment here is specifically whether it makes sense or not to add all the scaffolding (regardless of style) for single-case tests like these "not implemented" ones when the thing could just be:

```
func test_blahNotImplemented(t *testing.T) {
 	p, err := generateACME()
	assert.FatalError(t, err)
 	cert, opts, err := tc.p.AuthorizeSSHRekey(context.Background(), "")
 	assert.NotNil(t, err)
	assert.Nil(t, cert)
	assert.Nil(t, opts)
	sc, ok := err.(errs.StatusCoder)
	assert.Fatal(t, ok, "error does not implement StatusCoder interface")
	assert.Equals(t, sc.StatusCode(), http.StatusUnauthorized
	assert.HasPrefix(t, err.Error(), errors.New("provisioner.AuthorizeSSHRekey not implemented"))
}
```

Reply:
In general, I agree that all this scaffolding just for not implemented tests seems unnecessary. However, my expectation here is that we may be asked to implement these methods in the near future, so I wanted the scaffolding there so I wouldn't have to redo it later. 🤷‍♂

Reply:
Fair and it's already there so...